internal/fptest/portspeed.go

```diff
@@ -0,0 +1,43 @@
+// 2022 Nokia.  This code is a Contribution to OpenConfig Feature Profiles
+// project ("Work") made under the Google Software Grant and Corporate
+// Contributor License Agreement ("CLA") and governed by the Apache License 2.0.
+// No other rights or licenses in or to any of Nokia's intellectual property
+// are granted for any other purpose. This code is provided on an "as is" basis
+// without any warranties of any kind.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package fptest
+
+import (
+	"testing"
+
+	"github.com/openconfig/ondatra"
+	"github.com/openconfig/ondatra/gnmi"
+	"github.com/openconfig/ondatra/gnmi/oc"
+	"github.com/openconfig/ygot/ygot"
+)
+
+// Set Port Speed
+func SetPortSpeed(t *testing.T, d *ondatra.DUTDevice, p string) {
```

Comment:
```suggestion
func SetPortSpeed(t *testing.T, p *ondatra.Port) {
	dut, ok := p.Device().(*ondatra.DUTDevice)
	if !ok {
		t.Fatalf("Port is not from a DUT: %s", p)
	}
```

Reply:
Using `ondatra.Port` as suggested. However, `p.Device()` seems to return only device and no-error, thus not having suggested `if` condition.  Below function is called by the script - 

Ondatra `Device()` function - https://github.com/openconfig/ondatra/blob/f8de7c6fa979689f9c3f357a7e65b09504714a88/device.go#L160-L163

Reply:
You're right, gnmi.Update also accepts ondatra.Device, not just *ondatra.DUTDevice. Thanks!